### DISCUSSION TITLE
vips8.14.2 coredump in vips::VObject::VObject(vips::VObject const&) ()

### DISCUSSION THREAD

**Discussion by doudouodong** [libvips/libvips#3519] [General]
#0  0x000000000042e862 in vips::VObject::VObject(vips::VObject const&) ()
#1  0x000000000042ee0d in vips::VImage::VImage(vips::VImage const&) ()
#2  0x0000000000433f02 in ImagexxxxImpl::xxxx(google::protobuf::RpcController*, imagexxxx::ImageListRequest const*, imagerender::ImageListResponse*, google::protobuf::Closure*) ()
#3  0x000000000053a699 in imagerender::Imagexxx::CallMethod(google::protobuf::MethodDescriptor const*, google::protobuf::RpcController*, google::protobuf::Message const*, google::protobuf::Message*, google::protobuf::Closure*) ()
#4  0x0000000000663154 in brpc::policy::CallMethodWrappedWithCat(google::protobuf::Service*, google::protobuf::MethodDescriptor const*, google::protobuf::RpcController*, google::protobuf::Message const*, google::protobuf::Message*, google::protobuf::Closure*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, butil::EndPoint const&) ()
#5  0x0000000000664b81 in brpc::policy::ProcessHttpRequest(brpc::InputMessageBase*) ()
#6  0x00000000006acdd8 in brpc::ProcessInputMessage(void*) ()
#7  0x0000000000755805 in bthread::TaskGroup::task_runner(long) ()
#8  0x000000000077e9c1 in bthread_make_fcontext ()

**Comment by lovell** [comment]
Please provide considerably more information. This includes complete, standalone (e.g. no protobuf) code, enough that allows someone else to reproduce.

**Reply by doudouodong** [reply]
(gdb) bt
#0  0x00007f23656bc207 in raise () at /usr/lib64/libc.so.6
#1  0x00007f23656bd8f8 in abort () at /usr/lib64/libc.so.6
#2  0x00007f2366fd4725 in g_assertion_message () at /usr/lib64/libglib-2.0.so.0
#3  0x00007f2366fd478a in g_assertion_message_expr () at /usr/lib64/libglib-2.0.so.0
#4  0x00007f2366f7d42e in  () at /usr/lib64/libglib-2.0.so.0
#5  0x00007f2366f9d199 in g_hash_table_lookup () at /usr/lib64/libglib-2.0.so.0
#6  0x00007f2362a41f9d in _pango_fc_font_map_remove (fcfontmap=<optimized out>, fcfont=0x7f2328195520) at ../pango/pangofc-fontmap.c:1259
#7  0x00007f2362a3e1ed in pango_fc_font_finalize (object=0x7f2328195520) at ../pango/pangofc-font.c:161
#8  0x00007f23674d0a84 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#9  0x00007f2362fa17c7 in pango_item_free (item=0x7f2324161740) at ../pango/pango-item.c:104
#10 0x00007f2362fa2531 in free_run (run=0x7f2298031460, data=<optimized out>) at ../pango/pango-layout.c:2903
#11 0x00007f2366fcc328 in g_slist_foreach () at /usr/lib64/libglib-2.0.so.0
#12 0x00007f2362fa3711 in pango_layout_line_unref (line=0x7f22c80f2680) at ../pango/pango-layout.c:4066
#13 0x00007f2362fa375b in pango_layout_clear_lines (layout=layout@entry=0x7f216c110c50) at ../pango/pango-layout.c:2855
#14 0x00007f2362fa41f4 in pango_layout_finalize (object=0x7f216c110c50) at ../pango/pango-layout.c:225
#15 0x00007f23674d0a84 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#16 0x00007f2367967dd9 in vips_text_dispose (gobject=0x7f22940e9380) at ../libvips/create/text.c:144
#17 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#18 0x00007f23674d3a22 in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#19 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#20 0x00007f2367903c4a in vips_object_arg_close (argument=0x7f22c91b3020, argument_instance=0x7f2334764d80) at ../libvips/iofuncs/object.c:1078
#21 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#22 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#23 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#24 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#25 0x00007f23679028ea in vips_object_close (object=0x7f22c91b3020) at ../libvips/iofuncs/object.c:291
#26 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f22c91b3020) at ../libvips/iofuncs/object.c:1023
#27 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f22c91b3020) at ../libvips/iofuncs/image.c:545
#28 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#29 0x00007f23679036f3 in vips_object_clear_member (argument_instance=0x7f23340d0140) at ../libvips/iofuncs/object.c:890
#30 0x00007f2367903cf7 in vips__object_set_member (object=0x7f230c46a0a0, pspec=0x7f22c81372c0, member=0x7f230c46a108, argument=0x0) at ../libvips/iofuncs/object.c:1099
#31 0x00007f236790448d in vips_object_set_property (gobject=0x7f230c46a0a0, property_id=204, value=0x7f22b96f5e80, pspec=0x7f22c81372c0) at ../libvips/iofuncs/object.c:1255
#32 0x00007f23674d3bbb in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#33 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#34 0x00007f23679038ab in vips_object_dispose_argument (object=0x7f230c46a0a0, pspec=0x7f22c81372c0, argument_class=0x7f22c8139740, argument_instance=0x7f23340d0140, a=0x0, b=0x0) at ../libvips/iofuncs/object.c:928
#35 0x00007f23679030bb in vips_argument_map (object=0x7f230c46a0a0, fn=0x7f236790371d <vips_object_dispose_argument>, a=0x0, b=0x0) at ../libvips/iofuncs/object.c:606
#36 0x00007f23679038db in vips_argument_dispose_all (object=0x7f230c46a0a0) at ../libvips/iofuncs/object.c:954
#37 0x00007f2367903a6c in vips_object_dispose (gobject=0x7f230c46a0a0) at ../libvips/iofuncs/object.c:1021
#38 0x00007f236791c63f in vips_operation_dispose (gobject=0x7f230c46a0a0) at ../libvips/iofuncs/operation.c:240
#39 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#40 0x00007f23674d3a22 in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#41 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#42 0x00007f2367903c4a in vips_object_arg_close (argument=0x7f22d4daac80, argument_instance=0x7f23340d0210) at ../libvips/iofuncs/object.c:1078
#43 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#44 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#45 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#46 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#47 0x00007f23679028ea in vips_object_close (object=0x7f22d4daac80) at ../libvips/iofuncs/object.c:291
#48 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f22d4daac80) at ../libvips/iofuncs/object.c:1023
#49 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f22d4daac80) at ../libvips/iofuncs/image.c:545
#50 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#51 0x00007f2367908dbb in vips_object_local_cb (vobject=0x7f2334e9fe30, gobject=0x7f22d4daac80) at ../libvips/iofuncs/object.c:3052
#52 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#53 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#54 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#55 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#56 0x00007f23679028ea in vips_object_close (object=0x7f2334e9fe30) at ../libvips/iofuncs/object.c:291
#57 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f2334e9fe30) at ../libvips/iofuncs/object.c:1023
#58 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f2334e9fe30) at ../libvips/iofuncs/image.c:545
#59 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#60 0x00007f23679036f3 in vips_object_clear_member (argument_instance=0x7f2335130140) at ../libvips/iofuncs/object.c:890
#61 0x00007f2367903cf7 in vips__object_set_member (object=0x7f2218341800, pspec=0x7f22c8133af0, member=0x7f2218341868, argument=0x0) at ../libvips/iofuncs/object.c:1099
#62 0x00007f236790448d in vips_object_set_property (gobject=0x7f2218341800, property_id=177, value=0x7f22b96f6fc0, pspec=0x7f22c8133af0) at ../libvips/iofuncs/object.c:1255
#63 0x00007f23674d3bbb in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#64 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#65 0x00007f23679038ab in vips_object_dispose_argument (object=0x7f2218341800, pspec=0x7f22c8133af0, argument_class=0x7f22c8136d40, argument_instance=0x7f2335130140, a=0x0, b=0x0) at ../libvips/iofuncs/object.c:928
#66 0x00007f23679030bb in vips_argument_map (object=0x7f2218341800, fn=0x7f236790371d <vips_object_dispose_argument>, a=0x0, b=0x0) at ../libvips/iofuncs/object.c:606
#67 0x00007f23679038db in vips_argument_dispose_all (object=0x7f2218341800) at ../libvips/iofuncs/object.c:954
#68 0x00007f2367903a6c in vips_object_dispose (gobject=0x7f2218341800) at ../libvips/iofuncs/object.c:1021
#69 0x00007f236791c63f in vips_operation_dispose (gobject=0x7f2218341800) at ../libvips/iofuncs/operation.c:240
#70 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#71 0x00007f23674d3a22 in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#72 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#73 0x00007f2367903c4a in vips_object_arg_close (argument=0x7f23246bb7e0, argument_instance=0x7f2334ff5ec0) at ../libvips/iofuncs/object.c:1078
#74 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#75 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#76 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#77 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#78 0x00007f23679028ea in vips_object_close (object=0x7f23246bb7e0) at ../libvips/iofuncs/object.c:291
#79 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f23246bb7e0) at ../libvips/iofuncs/object.c:1023
#80 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f23246bb7e0) at ../libvips/iofuncs/image.c:545
#81 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#82 0x00007f23679036f3 in vips_object_clear_member (argument_instance=0x7f23341bcb40) at ../libvips/iofuncs/object.c:890
#83 0x00007f2367903cf7 in vips__object_set_member (object=0x7f220c097ba0, pspec=0x7f22c8133af0, member=0x7f220c097c08, argument=0x0) at ../libvips/iofuncs/object.c:1099
#84 0x00007f236790448d in vips_object_set_property (gobject=0x7f220c097ba0, property_id=177, value=0x7f22b96f7c20, pspec=0x7f22c8133af0) at ../libvips/iofuncs/object.c:1255
#85 0x00007f23674d3bbb in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#86 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#87 0x00007f23679038ab in vips_object_dispose_argument (object=0x7f220c097ba0, pspec=0x7f22c8133af0, argument_class=0x7f22c8136d40, argument_instance=0x7f23341bcb40, a=0x0, b=0x0) at ../libvips/iofuncs/object.c:928
#88 0x00007f23679030bb in vips_argument_map (object=0x7f220c097ba0, fn=0x7f236790371d <vips_object_dispose_argument>, a=0x0, b=0x0) at ../libvips/iofuncs/object.c:606
#89 0x00007f23679038db in vips_argument_dispose_all (object=0x7f220c097ba0) at ../libvips/iofuncs/object.c:954
#90 0x00007f2367903a6c in vips_object_dispose (gobject=0x7f220c097ba0) at ../libvips/iofuncs/object.c:1021
#91 0x00007f236791c63f in vips_operation_dispose (gobject=0x7f220c097ba0) at ../libvips/iofuncs/operation.c:240
#92 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#93 0x00007f23674d3a22 in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#94 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#95 0x00007f2367903c4a in vips_object_arg_close (argument=0x7f2324f23e30, argument_instance=0x7f23341b8280) at ../libvips/iofuncs/object.c:1078
#96 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#97 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#98 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#99 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#100 0x00007f23679028ea in vips_object_close (object=0x7f2324f23e30) at ../libvips/iofuncs/object.c:291
#101 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f2324f23e30) at ../libvips/iofuncs/object.c:1023
#102 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f2324f23e30) at ../libvips/iofuncs/image.c:545
#103 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#104 0x00007f2367908dbb in vips_object_local_cb (vobject=0x7f22d4253980, gobject=0x7f2324f23e30) at ../libvips/iofuncs/object.c:3052
#105 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#106 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#107 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#108 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#109 0x00007f23679028ea in vips_object_close (object=0x7f22d4253980) at ../libvips/iofuncs/object.c:291
#110 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f22d4253980) at ../libvips/iofuncs/object.c:1023
#111 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f22d4253980) at ../libvips/iofuncs/image.c:545
#112 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#113 0x00007f2367908e1a in vips_object_local_array_cb (parent=0x7f22c41a1410, local=0x7f2334ea4340) at ../libvips/iofuncs/object.c:3066
#114 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#115 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#116 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#117 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#118 0x00007f23679028ea in vips_object_close (object=0x7f22c41a1410) at ../libvips/iofuncs/object.c:291
#119 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f22c41a1410) at ../libvips/iofuncs/object.c:1023
#120 0x00007f236791c63f in vips_operation_dispose (gobject=0x7f22c41a1410) at ../libvips/iofuncs/operation.c:240
#121 0x00007f2367875f6b in vips_composite_base_dispose(GObject*) (gobject=0x7f22c41a1410) at ../libvips/conversion/composite.cpp:163
#122 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#123 0x00007f23674d3a22 in g_object_set_valist () at /usr/lib64/libgobject-2.0.so.0
#124 0x00007f23674d45bc in g_object_set () at /usr/lib64/libgobject-2.0.so.0
#125 0x00007f2367903c4a in vips_object_arg_close (argument=0x7f226c0a8340, argument_instance=0x7f23341a9140) at ../libvips/iofuncs/object.c:1078
#126 0x00007f23674cba18 in g_closure_invoke () at /usr/lib64/libgobject-2.0.so.0
#127 0x00007f23674de0fd in  () at /usr/lib64/libgobject-2.0.so.0
#128 0x00007f23674e6091 in g_signal_emit_valist () at /usr/lib64/libgobject-2.0.so.0
#129 0x00007f23674e637f in g_signal_emit () at /usr/lib64/libgobject-2.0.so.0
#130 0x00007f23679028ea in vips_object_close (object=0x7f226c0a8340) at ../libvips/iofuncs/object.c:291
#131 0x00007f2367903a78 in vips_object_dispose (gobject=0x7f226c0a8340) at ../libvips/iofuncs/object.c:1023
#132 0x00007f236790ae5a in vips_image_dispose (gobject=0x7f226c0a8340) at ../libvips/iofuncs/image.c:545
#133 0x00007f23674d0992 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#134 0x000000000042f337 in vips::VObject::~VObject() ()
#135 0x000000000042f57a in vips::VImage::~VImage() ()
#136 0x000000000043afeb in void std::_Destroy<vips::VImage>(vips::VImage*) ()
#137 0x000000000043a893 in void std::_Destroy_aux<false>::__destroy<vips::VImage*>(vips::VImage*, vips::VImage*) ()
#138 0x0000000000439ace in void std::_Destroy<vips::VImage*>(vips::VImage*, vips::VImage*) ()
#139 0x0000000000438917 in void std::_Destroy<vips::VImage*, vips::VImage>(vips::VImage*, vips::VImage*, std::allocator<vips::VImage>&) ()
#140 0x00000000004374a9 in std::vector<vips::VImage, std::allocator<vips::VImage> >::~vector() ()
#141 0x0000000000434d19 in ImageRenderImpl::renderList(google::protobuf::RpcController*, imagerender::ImageListRequest const*, imagerender::ImageListResponse*, google::protobuf::Closure*) ()
#142 0x000000000053b183 in imagerender::ImageRender::CallMethod(google::protobuf::MethodDescriptor const*, google::protobuf::RpcController*, google::protobuf::Message const*, google::protobuf::Message*, google::protobuf::Closure*) ()
#143 0x0000000000663c3e in brpc::policy::CallMethodWrappedWithCat(google::protobuf::Service*, google::protobuf::MethodDescriptor const*, google::protobuf::RpcController*, google::protobuf::Message const*, google::protobuf::Message*, google::protobuf::Closure*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, butil::EndPoint const&) ()
#144 0x000000000066566b in brpc::policy::ProcessHttpRequest(brpc::InputMessageBase*) ()
#145 0x00000000006ad8c2 in brpc::ProcessInputMessage(void*) ()
#146 0x00000000006afd58 in brpc::RunLastMessage::operator()(brpc::InputMessageBase*) ()
#147 0x00000000006afdd7 in std::unique_ptr<brpc::InputMessageBase, brpc::RunLastMessage>::~unique_ptr() ()
#148 0x00000000006ae5bb in brpc::InputMessenger::OnNewMessages(brpc::Socket*) ()
#149 0x0000000000593073 in brpc::Socket::ProcessEvent(void*) ()
#150 0x00000000007562ef in bthread::TaskGroup::task_runner(long) ()
#151 0x000000000077f4b1 in bthread_make_fcontext ()

**Reply by doudouodong** [reply]
g_object_unref  frequently  called

**Comment by jcupitt** [comment]
Hi @doudouodong,

I would guess you are mixing up C++ compiler versions, or perhaps you have made an error in reference counting. It might also be a threading issue.

You need to make a small, complete program which will make it possible for one of us reproduce this crash. You must also include:

- compiler version
- libvips version
- if you are using a precompiled libvips binary, explain where that came from
- platform and platform version
- explain how you built and linked the program
